Vega to SVG conversion failure on VegaFusion CI

VegaFusion CI is failing with:

```
---------------------------- Captured stderr call -----------------------------
thread '<unnamed>' panicked at 'from_rgba returned None', vl-convert-rs\src\converter.rs:817:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
___________________ test_it[custom-binned_scatter.vg.json] ____________________

category = 'custom', name = 'binned_scatter.vg.json'

    @pytest.mark.duckdb
    @pytest.mark.parametrize(
        "category,name", load_test_cases())
    def test_it(category, name):
    
        maybe_skip(category, name)
    
        # Load spec into dict
        file = spec_dir / category / name
        spec = json.loads(file.read_text("utf8"))
    
        # Define local timezone
        local_tz = "America/New_York"
    
        # Pre-transform with DataFusion connection and convert to image
        vf.runtime.set_connection("datafusion")
        (transformed, _) = vf.runtime.pre_transform_spec(spec, local_tz)
>       img_datafusion = imread(BytesIO(vega_to_png(transformed)))
E       ValueError: Vega to SVG conversion failed:
E       Err(Any { .. })
```

This is logic that was copied and modified from tiny_skia to make it possible to set the PNG ppi. I'll investigate

--- a/vl-convert-rs/src/converter.rs
+++ b/vl-convert-rs/src/converter.rs
@@ -863,7 +863,8 @@ pub fn encode_png(pixmap: Pixmap, ppi: f32) -> Result<Vec<u8>, AnyError> {
         let alpha = c.alpha();
 
         // jonmmease: tiny-skia uses the private PremultipliedColorU8::from_rgba_unchecked here,
-        // but all this method does is construct a `PremultipliedColorU8` from components.
+        // but we need to use from_rgba, which checks to make sure r/g/b are less then or equal
+        // to alpha. Use min to ensure we don't trigger the check
         *pixel = PremultipliedColorU8::from_rgba(
             c.red().min(alpha),
             c.green().min(alpha),

--- a/vl-convert-rs/src/converter.rs
+++ b/vl-convert-rs/src/converter.rs
@@ -860,10 +860,16 @@ pub fn encode_png(pixmap: Pixmap, ppi: f32) -> Result<Vec<u8>, AnyError> {
     // due to rounding. So we stick with this method for now.
     for pixel in pixmap.pixels_mut() {
         let c = pixel.demultiply();
+        let alpha = c.alpha();
 
         // jonmmease: tiny-skia uses the private PremultipliedColorU8::from_rgba_unchecked here,
         // but all this method does is construct a `PremultipliedColorU8` from components.
-        *pixel = PremultipliedColorU8([c.red(), c.green(), c.blue(), c.alpha()]);
+        *pixel = PremultipliedColorU8::from_rgba(
+            c.red().min(alpha),
+            c.green().min(alpha),
+            c.blue().min(alpha),
+            alpha,
+        ).expect("Failed to construct PremultipliedColorU8 from rgba");
     }
 
     let mut data = Vec::new();

--- a/vl-convert-rs/src/converter.rs
+++ b/vl-convert-rs/src/converter.rs
@@ -869,7 +869,8 @@ pub fn encode_png(pixmap: Pixmap, ppi: f32) -> Result<Vec<u8>, AnyError> {
             c.green().min(alpha),
             c.blue().min(alpha),
             alpha,
-        ).expect("Failed to construct PremultipliedColorU8 from rgba");
+        )
+        .expect("Failed to construct PremultipliedColorU8 from rgba");
     }
 
     let mut data = Vec::new();

--- a/vl-convert-rs/src/converter.rs
+++ b/vl-convert-rs/src/converter.rs
@@ -860,8 +860,10 @@ pub fn encode_png(pixmap: Pixmap, ppi: f32) -> Result<Vec<u8>, AnyError> {
     // due to rounding. So we stick with this method for now.
     for pixel in pixmap.pixels_mut() {
         let c = pixel.demultiply();
-        *pixel = PremultipliedColorU8::from_rgba(c.red(), c.green(), c.blue(), c.alpha())
-            .expect("from_rgba returned None");
+
+        // jonmmease: tiny-skia uses the private PremultipliedColorU8::from_rgba_unchecked here,
+        // but all this method does is construct a `PremultipliedColorU8` from components.
+        *pixel = PremultipliedColorU8([c.red(), c.green(), c.blue(), c.alpha()]);
     }
 
     let mut data = Vec::new();

